fiftyone/core/models.py

```diff
@@ -487,18 +516,29 @@ def _apply_image_model_data_loader(
                     )
                     ctx.save(sample)
 
-            except Exception as e:
-                if not skip_failures:
-                    raise e
+        with _async_executor(
```

Comment:
You could avoid one level of indentation here by continuing the `context.enter_context()` pattern used above:

```py
submit = context.enter_context(_async_executor(...))

for sample_batch, imgs in zip(...):
    ...
```

Note that there's a concrete reason that `context.enter_context()` is used here: the [corresponding code in `fiftyone-teams`](https://github.com/voxel51/fiftyone-teams/blob/de11d3951876e86d00544a9c41d324478ff64473/fiftyone/core/models.py#L479-L498) contains an extra context that is unique to processing cloud data, and writing the code in that way minimizes merge conflicts when pulling changes in this code area over to `fiftyone-teams`.